src/Catalyst.Modules.Server.Blazor/AutofacServiceProvider.cs

```diff
@@ -0,0 +1,137 @@
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using Autofac;
+using Autofac.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+
+namespace Catalyst.Modules.Server.Blazor
+{
+    /// <summary>
+    /// A factory for creating a <see cref="ContainerBuilder"/> and an <see cref="IServiceProvider" />.
+    /// </summary>
+    public class AutofacServiceProviderFactory : IServiceProviderFactory<ContainerBuilder>
+    {
+        private readonly ContainerBuilder _containerBuilder;
+        private AutofacServiceProvider _autofacServiceProvider;
+
+        public AutofacServiceProviderFactory(ContainerBuilder containerBuilder)
+        {
+            _containerBuilder = containerBuilder;
+        }
+
+        /// <summary>
+        /// Creates a container builder from an <see cref="IServiceCollection" />.
+        /// </summary>
+        /// <param name="services">The collection of services.</param>
+        /// <returns>A container builder that can be used to create an <see cref="IServiceProvider" />.</returns>
+        public ContainerBuilder CreateBuilder(IServiceCollection services)
+        {
+            _containerBuilder.Populate(services);
+            return _containerBuilder;
+        }
+
+        public void SetContainer(IContainer container) { _autofacServiceProvider.LifetimeScope = container; }
+
+        /// <summary>
+        /// Creates an <see cref="IServiceProvider" /> from the container builder.
+        /// </summary>
+        /// <param name="containerBuilder">The container builder.</param>
+        /// <returns>An <see cref="IServiceProvider" />.</returns>
+        public IServiceProvider CreateServiceProvider(ContainerBuilder containerBuilder)
+        {
+            if (containerBuilder == null) throw new ArgumentNullException(nameof(containerBuilder));
+
+            return _autofacServiceProvider = new AutofacServiceProvider();
+        }
+    }
+
+    /// <summary>
+    /// Autofac implementation of the ASP.NET Core <see cref="IServiceProvider"/>.
+    /// </summary>
+    /// <seealso cref="System.IServiceProvider" />
+    /// <seealso cref="Microsoft.Extensions.DependencyInjection.ISupportRequiredService" />
+    internal class AutofacServiceProvider : IServiceProvider, ISupportRequiredService, IDisposable
+    {
+        public ILifetimeScope LifetimeScope { get; set; }
+
+        private bool _disposed = false;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this initialization to '_disposed', the compiler will do that for you.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4594394)